src/semantic/type_checker.rs

```diff
@@ -25,14 +24,24 @@ impl TypeEnv {
                 let left_type = self.type_of_expr(l)?;
                 let right_type = self.type_of_expr(r)?;
                 match (left_type, right_type, op) {
-                    (Type::Int, Type::Int, Operator::Plus) => Ok(Type::Int),
+                    (
+                        Type::ClassType { name: n1, .. },
+                        Type::ClassType { name: n2, .. },
+                        Operator::Plus,
+                    ) => {
+                        if n1.as_str() == "int" && n1 == n2 {
```

Comment:
Should model binary expression as a method, then we even don't need these codes.